ProcessMessage/handler.ts

```diff
@@ -422,6 +432,55 @@ export const getProcessMessageHandler = ({
                 rejection_reason: RejectionReasonEnum.USER_NOT_FOUND,
                 status: RejectedMessageStatusValueEnum.REJECTED
               }),
+              TE.chain(() =>
+                pipe(
+                  // setting TTL to 3 years for message-status entries
+                  lMessageStatusModel.updateTTLForAllVersions(
+                    [newMessageWithoutContent.id],
+                    TTL_FOR_USER_NOT_FOUND
+                  ),
+                  TE.mapLeft((error: CosmosErrors) => {
+                    telemetryClient.trackEvent({
+                      name: "api.messages.create.failstatusttlset",
+                      properties: {
+                        error: `Something went wrong trying to update the ttl for the message status | ${error.kind}`,
+                        fiscalCode: toHash(newMessageWithoutContent.fiscalCode),
+                        messageId: newMessageWithoutContent.id,
+                        senderId: newMessageWithoutContent.senderServiceId
+                      },
+                      tagOverrides: { samplingEnabled: "false" }
+                    });
+                    return error;
+                  })
+                )
+              ),
+              TE.chain(() =>
+                pipe(
+                  // setting TTL to 3 years for message entry
+                  lMessageModel.patch(
+                    [
+                      newMessageWithoutContent.id,
+                      newMessageWithoutContent.fiscalCode
+                    ],
+                    { ttl: TTL_FOR_USER_NOT_FOUND } as Partial<
+                      MessageWithoutContent
+                    >
```

Comment:
We know that `ttl` is not allowed by patch, even if it's correctly working at runtime. Maybe it's better to add a little comment to explain why we need this cast.